app/helpers/pipeline_runs_helper.rb

```diff
@@ -0,0 +1,96 @@
+require 'open3'
+
+module PipelineRunsHelper
+  def aegea_batch_submit_command(base_command,
+                                 memory = Sample::DEFAULT_MEMORY_IN_MB,
+                                 job_queue: nil,
+                                 docker_image: "idseq_dag")
+    command = "aegea batch submit --command=\"#{base_command}\" "
+    if memory <= Sample::DEFAULT_MEMORY_IN_MB
+      vcpus = Sample::DEFAULT_VCPUS
+      queue = Sample::DEFAULT_QUEUE
+    else
+      vcpus = Sample::DEFAULT_VCPUS_HIMEM
+      queue = Sample::DEFAULT_QUEUE_HIMEM
+    end
+    if job_queue.present?
+      if Sample::DEPRECATED_QUEUES.include? job_queue
+        Rails.logger.info "Overriding deprecated queue #{job_queue} with #{queue}"
+      else
+        queue = job_queue
+      end
+    end
+    command += " --storage /mnt=#{Sample::DEFAULT_STORAGE_IN_GB} --volume-type gp2 --ecr-image #{docker_image} --memory #{memory} --queue #{queue} --vcpus #{vcpus} --job-role idseq-pipeline "
+    command
+  end
+
+  def job_info(job_id, run_id)
+    job_status = nil
+    job_log_id = nil
+    job_hash = nil
+    stdout, stderr, status = Open3.capture3("aegea", "batch", "describe", job_id.to_s)
+    if status.exitstatus.zero?
+      job_description = stdout
+      job_hash = JSON.parse(job_description)
+      job_status = job_hash['status']
+      if job_hash['container'] && job_hash['container']['logStreamName']
+        job_log_id = job_hash['container']['logStreamName']
+      end
+    else
+      Airbrake.notify("Error for update job status for pipeline run #{run_id} with error #{stderr}")
+      job_status = STATUS_ERROR # transient error, job is still "in progress"
+      job_status = STATUS_FAILED if stderr =~ /IndexError/ # job no longer exists
+    end
+    [job_status, job_log_id, job_hash, stdout]
+  end
+
+  def file_generated_since_run(record, s3_path)
+    stdout, _stderr, status = Open3.capture3("aws", "s3", "ls", s3_path.to_s)
+    return false unless status.exitstatus.zero?
+    begin
+      s3_file_time = DateTime.strptime(stdout[0..18], "%Y-%m-%d %H:%M:%S")
+      return (s3_file_time && record.created_at && s3_file_time > record.created_at)
+    rescue
+      return nil
+    end
+  end
+
+  def update_pipeline_version(record, version_column, s3_version_file)
```

Comment:
Orthogonal to this pull request, but this is an awkward way to get the version.   When we dispatch a job, we know what version we are dispatching.  Extract the version there and store it;  then we don't have to poll and look if the timestamp is current.

Reply:
Actually when we dispatch the job we don't know the version yet. As far as I know, we need to clone idseq-dag before we can check for the version. But yes, this has always felt awkward. If there were a way to just read a single file from github, that would allow us to easily get the version immediately upon dispatch rather than cloning the whole repo in the worker.

Reply:
Worthy of an issue.   I think there is a way to read a single file.  It's also possible to run the idseq-dag command with argument that is its githash, and then it would dump the version into a file named with the githash, so we don't have to check its timestamp.  But that's also cumbersome.